src/SDK.ts

```diff
@@ -642,21 +642,38 @@ export default class SDK implements ISDK {
     });
 
     const { reconnectId, authenticatedUserToken, initContext, getContext } = sessionInitOptionalParams;
-    const requestHeaders: StringMap = Constants.defaultHeaders;
-    let requestPath = `/${OmnichannelEndpoints.LiveChatSessionInitPath}/${this.omnichannelConfiguration.orgId}/${this.omnichannelConfiguration.widgetId}/${requestId}`;
-    if (authenticatedUserToken) {
-      requestPath = `/${OmnichannelEndpoints.LiveChatAuthSessionInitPath}/${this.omnichannelConfiguration.orgId}/${this.omnichannelConfiguration.widgetId}/${requestId}`;
-      requestHeaders[OmnichannelHTTPHeaders.authenticatedUserToken] = authenticatedUserToken;
-      requestHeaders[OmnichannelHTTPHeaders.authCodeNonce] = this.configuration.authCodeNonce;
+    const data: InitContext = initContext || {};
+    const requestHeaders: StringMap = { ...Constants.defaultHeaders };
+
+    const isV2 = useSessionInitV2;
+    const basePath = authenticatedUserToken
+      ? isV2
+        ? OmnichannelEndpoints.LiveChatSessionInitAuthChatV2Path
+        : OmnichannelEndpoints.LiveChatAuthSessionInitPath
+      : isV2
+        ? OmnichannelEndpoints.LiveChatSessionInitV2Path
+        : OmnichannelEndpoints.LiveChatSessionInitPath;
+
+    let requestPath = isV2
+    ? `/${basePath}/${this.omnichannelConfiguration.orgId}/widgetApp/${this.omnichannelConfiguration.widgetId}/conversation`
+    :`/${basePath}/${this.omnichannelConfiguration.orgId}/${this.omnichannelConfiguration.widgetId}/${requestId}`;
+    
+
+    if (authenticatedUserToken) {     
+      Object.assign(requestHeaders, {
+        [OmnichannelHTTPHeaders.authenticatedUserToken]: authenticatedUserToken,
+        [OmnichannelHTTPHeaders.authCodeNonce]: this.configuration.authCodeNonce,
+      });
```

Comment:
can we abstract v2 code to a separate function?, at least the setup, to ennhance readability

Reply:
You can reuse [endpointsCreators.ts](https://github.com/microsoft/omnichannel-sdk/blob/main/src/Utils/endpointsCreators.ts)